src/pam_google_authenticator.c

```diff
@@ -2132,6 +2174,13 @@ static int google_authenticator(pam_handle_t *pamh,
                   "but can't switch back", old_uid, uid);
     }
   }
+  // restore old supplementary groups for process
+  if (old_groups.n_groups > 0) {
```

Comment:
If the list is empty we still want to call `setgroups()`, no? That is, we want to drop whatever groups we added before.

Reply:
I think when we make sure that the effective group is taken into account, then the list can never be empty. Except for platforms where not all functions are available. I will make sure that this case is also handled correctly.